assets/src/components/supported-templates-toggle/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Options } from '../options-context-provider';
+import { AMPSettingToggle } from '../amp-setting-toggle';
+import { AMPNotice } from '../amp-notice';
+
+export function SupportedTemplatesToggle( { themeSupportArgs } ) {
+	const { editedOptions, updateOptions } = useContext( Options );
+
+	const { all_templates_supported: allTemplatesSupported, theme_support: themeSupport } = editedOptions;
+
+	return themeSupportArgs && 'templates_supported' in themeSupportArgs && 'all' === themeSupportArgs.templates_supported
+		? (
+			<AMPNotice>
+				<p>
+					{ __( 'The current theme requires all templates support AMP.', 'amp' ) }
```

Comment:
If the theme has `templates_supported => 'all'`, maybe we shouldn't show the "Serve all templates as AMP" toggle at all. @westonruter Would this make sense?

Reply:
I was actually thinking of eliminating the `templates_supported` option together. It doesn't really make sense because you can still select Transitional mode, and that means that all templates _will_ be accessible without AMP anyway.

In spite of this, when this theme support flag is present, the post editor screams at you that AMP can't be disabled:

![image](https://user-images.githubusercontent.com/134745/87000538-2504ff00-c16a-11ea-85dd-feddcc28aba5.png)

This was raised in #1864. 

This was originally put in place when we wanted a site to have `add_theme_support('amp')` mean that Standard mode was _forced_, and then the `themes_supported => all` would then make sure that no templates could be disabled. This is too draconian, however.

Also discussed in #2724.

No themes on WordPress.org use this theme support flag. While the flag is being used by @swissspidy, if we just remove it then it just means the UI to select templates will be available, but they'll still all be enabled by default.

So I think we should eliminate `themes_supported` altogether.

Reply:
This is related to what I mentioned in https://github.com/ampproject/amp-wp/pull/4998#discussion_r451870597 with defaulting to `paired => true` when `paired` is absent. The reality is that all themes need to work when AMP is turned off, so that means all themes need to support paired.

At the moment this is painful since it can mean functionality needs AMP and non-AMP implementations. But when Bento AMP lands, this will be easy.

Reply:
This also means eliminating the current logic in `\AMP_Theme_Support::get_supportable_templates()` for `supported` and `immutable`.

Reply:
I also think the `amp` post type support flag should be eliminated in favor of just using the DB options. This would eliminate the need for the logic in `AMP_Options_Manager::check_supported_post_type_update_errors()` and a mess of trying to respect DB-configuration as well as DB-specified post type support.

If anything, the `amp` post type support should indicate whether AMP is enabled by _default_. But it should always be overridable in the admin.

Reply:
Started working on that in #5010.

Reply:
Bottom line: for now just be concerned with what is reflected in the DB option.

Reply:
👍 Removed the `themeSupportArgs && 'all' === themeSupportArgs.templates_supported` check entirely in 4881c2da2.